src/main/java/io/cryostat/targets/AgentConnection.java

```diff
@@ -86,7 +91,11 @@ public String getHost() {
     }
 
     public static boolean isAgentConnection(URI uri) {
-        return Set.of("http", "https", "cryostat-agent").contains(uri.getScheme());
+        if (TLS_ENABLED) {
```

Comment:
I'm not sure if this check is quite right. In the case that Agent TLS is enabled (or maybe "required" is better terminology), then an `http://` connection URL should never have gotten into the system at all to make it to this point. But if it did somehow, then we still should be able to recognize that this is an Agent connection, as opposed to a JMX connection.